bundle/src/main/java/com/adobe/acs/commons/dispatcher/impl/PermissionSensitiveCacheServlet.java

```diff
@@ -0,0 +1,121 @@
+
+@Component(
+        label = "ACS AEM Commons - Permission Sensitive Cache Servlet",
+        description = "Servlet that checks if the current sessions has access to a cached object",
+        metatype = true, immediate = true
+)
+@Properties({
+        @Property(
+                label = "File extensions to be removed from the URI",
+                name = REMOVEABLE_EXTENSIONS,
+                description = "Extensions that can be removed from the URI. For example, .html",
+                cardinality = Integer.MAX_VALUE,
+                value = {"html"}
+        ),
+        @Property(
+                name = "sling.servlet.paths",
+                cardinality = Integer.MAX_VALUE,
+                label = "Sling Servlet Paths",
+                description = "Paths that this servlet will resolve to"
+        )
+})
+@Service
+public class PermissionSensitiveCacheServlet extends SlingSafeMethodsServlet {
+
+    private final Logger log = LoggerFactory.getLogger(PermissionSensitiveCacheServlet.class);
+    public static final String REMOVEABLE_EXTENSIONS = "removable-extensions";
+
+    private List<String> removableExtensions = new ArrayList<String>();
+
+    @Activate
+    public final void activate(final Map<String, String> config) {
+
+        String[] propertyArray = PropertiesUtil.toStringArray(config.get(REMOVEABLE_EXTENSIONS),
+                new String[]{"html"});
+        removableExtensions = Arrays.asList( propertyArray );
+    }
+
+    public void doHead(SlingHttpServletRequest request, SlingHttpServletResponse response) {
+        doHeadRequest(request, response);
+    }
+
+    public void doHeadRequest(SlingHttpServletRequest request, SlingHttpServletResponse response) {
+        try{
+            //retrieve the requested URL
+            String requestUri = request.getParameter( "uri" );
+            log.debug( "Request URI {}", requestUri );
+
+            if(StringUtils.isNotEmpty( requestUri ) ){
+
+                String uri = getUri( requestUri );
+
+                log.info( "Checking permissions for {}", uri );
+
+                Resource requestedResource = request.getResourceResolver().getResource( uri );
+
+                if (requestedResource != null) {
+                    log.debug("Current Session has access to {}", uri );
+                    response.setStatus(SlingHttpServletResponse.SC_OK);
+                } else {
+                    log.debug("Current Session does not have access to {}", uri );
+                    response.setStatus(SlingHttpServletResponse.SC_UNAUTHORIZED);
+                }
+            } else {
+                response.setStatus(SlingHttpServletResponse.SC_UNAUTHORIZED);
+            }
+        } catch(Exception e){
+            log.error("authchecker servlet exception: " + e.getMessage());
+        }
+    }
+
+    public String getUri(String requestUri){
+
+        String extension = FilenameUtils.getExtension( requestUri );
```

Comment:
Using `FilenameUtils` feels wrong in that the param is  URI and not a file... If we cant just user` rr.resolve(..)` (described above) which makes this trimming moot, how about using `PathInfo` to parse the URI at which point you have direct access to the resource path (`pathInfo.getResourcePath()`) ... 

https://docs.adobe.com/docs/en/aem/6-3/develop/ref/javadoc/com/day/cq/commons/PathInfo.html